src/libstd/sys/unix/condvar.rs

```diff
@@ -46,33 +47,40 @@ impl Condvar {
         debug_assert_eq!(r, 0);
     }
 
+    // This implementation is modeled after libcxx's condition_variable
     pub unsafe fn wait_timeout(&self, mutex: &Mutex, dur: Duration) -> bool {
-        assert!(dur >= Duration::nanoseconds(0));
+        if dur <= Duration::zero() {
+            return false;
+        }
```

Comment:
I'm not sure how semantically significant this is, but this means that the lock is never unlocked. It seems pretty minor though.